tests/scripts/multi-node/build-rook.sh

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env bash
+set -e
+
+
+#############
+# VARIABLES #
+#############
+
+rook_git_root=$(git rev-parse --show-toplevel)
+rook_kube_templates_dir="$rook_git_root/cluster/examples/kubernetes/"
+
+
+#############
+# FUNCTIONS #
+#############
+
+function fail_if {
+  if [[ "$(uname -n)" != "k8s-01" ]]; then
+    echo "You must run the script from k8s-01"
+    exit 1
+  fi
+  if [[ "$(id -nu)" != "vagrant" ]]; then
+    echo "Wrong user, log as 'vagrant'"
+    exit 1
+  fi
+  if ! git rev-parse --show-toplevel &> /dev/null; then
+    echo "It looks like you are NOT in a Git repository"
+    echo "This script should be executed from WITHIN Rook's git repository"
+    exit 1
+  fi
+}
+
+function purge_rook_pods {
+  cd "$rook_kube_templates_dir"
+  kubectl delete -n rook pool replicapool || true
+  kubectl delete storageclass rook-block || true
+  kubectl -n kube-system delete secret rook-admin || true
+  kubectl delete -f kube-registry.yaml || true
+  kubectl delete -n rook cluster rook || true
+  kubectl delete thirdpartyresources cluster.rook.io pool.rook.io objectstore.rook.io filesystem.rook.io volumeattachment.rook.io || true # ignore errors if on K8s 1.7+
+  kubectl delete crd clusters.rook.io pools.rook.io objectstores.rook.io filesystems.rook.io volumeattachments.rook.io || true # ignore errors if on K8s 1.5 and 1.6
+  kubectl delete -n rook-system daemonset rook-agent || true
+  kubectl delete -f rook-operator.yaml || true
+  kubectl delete clusterroles rook-agent || true
+  kubectl delete clusterrolebindings rook-agent || true
+  kubectl delete namespace rook || true
+  cd "$rook_git_root"
+}
+
+function install_rpm_dep {
+  sudo yum install -y go git perl-Digest-SHA
+}
+
+  # shellcheck disable=SC2120
+function add_user_to_docker_group {
+  sudo groupadd docker || true
+  sudo gpasswd -a vagrant docker || true
+  if [[ $(id -gn) != docker ]]; then
+    exec sg docker "$0 $*"
+  fi
+}
+
+function run_docker_registry {
+  if ! docker ps | grep -sq registry; then
+    docker run -d -p 5000:5000 --restart=always --name registry registry:2
+  fi
+}
+
+function docker_import {
+  img=$(docker images | grep -Eo '^build-[a-z0-9]{8}/rook-[a-z0-9]+\s')
+  # shellcheck disable=SC2086
+  docker tag $img 172.17.8.1:5000/rook/rook:latest
```

Comment:
Vagrant's VMs are running at addresses 100+i, i in 0..n, so I assume this should be "172.17.8.100"?

Reply:
Leftover from my previous attempt where the registry is hosted on the host itself, not inside the first VM (k8s-01). So this is still valid.